scanner/Makefile

```diff
@@ -34,10 +34,12 @@ endif
 GO_VERSION := $(firstword $(subst go version ,,$(shell go version)))
 EXPECTED_GO_VERSION := $(shell cat ../EXPECTED_GO_VERSION)
 
-BUILD_FLAGS := CGO_ENABLED=0 GOOS=${GOOS} GOARCH=${GOARCH}
-BUILD_CMD := $(BUILD_FLAGS) go build \
+
+GO_BUILD_FLAGS := CGO_ENABLED=0 GOOS=${GOOS} GOARCH=${GOARCH}
```

Comment:
```suggestion
GO_BUILD_FLAGS := CGO_ENABLED=0 GOOS=${HOST_OS} GOARCH=${GOARCH}
```

Reply:
Needed to run on mac, believe this makes the `GOOS` initialization on line `28` unused.

Reply:
That would break the image build. If you call make GOOS='$(HOST_OS)' you should be able to get a build for your local OS.

Reply:
Not sure I follow why this break the build for scanner, but doesn't break the build for the other ACS components that do the same thing?

Having to specify the `GOOS` every time running `make` locally seems like an unnecessary pain - changing the default to be dynamic still allows for CI or other usages to override it via `make GOOS=...`

Reply:
> Not sure I follow why this break the build for scanner, but doesn't break the build for the other ACS components that do the same thing?

If you're building the image locally, you will want to use the `GOOS=DEFAULT_GOOS`. If you set `GOOS=${HOST_OS}` and `${HOST_OS} != ${DEFAULT_GOOS}`, then building the image locally would create a binary that is incompatible with the container AFAIK, since we copy the binary into the image when building it locally.

> Having to specify the GOOS every time running make locally seems like an unnecessary pain

You only have to specify `GOSS` if your `HOST_OS` is different from `DEFAULT_GOOS`. As a convenience, there is a target in the `Makefile` for these cases when you're building the scanner for running it in your machine, `scanner-go-build-local` -- it essentially replaces the `GOOS` accordingly.

> - changing the default to be dynamic still allows for CI or other usages to override it via make HOST_OS=...

The default favors the image build instead of the binary for local development.

Reply:
This conversation is essential and takes us to discuss the most maintainable and reliable way to build a scanner. Still, for E2E tests, it is orthogonal. We should not block this PR and move it elsewhere if you agree.